sklearn/metrics/tests/test_pairwise.py

```diff
@@ -272,8 +247,30 @@ def callable_rbf_kernel(x, y, **kwds):
          (pairwise_distances, 'wminkowski', _wminkowski_kwds),
          (pairwise_kernels, 'polynomial', {'degree': 1}),
          (pairwise_kernels, callable_rbf_kernel, {'gamma': .1})])
-def test_pairwise_parallel(func, metric, kwds):
-    check_pairwise_parallel(func, metric, kwds)
+@pytest.mark.parametrize('array_constr', [np.array, csr_matrix])
+@pytest.mark.parametrize('dtype', [np.float64, int])
+def test_pairwise_parallel(func, metric, kwds, array_constr, dtype):
+    rng = np.random.RandomState(0)
+    X = array_constr(5 * rng.random_sample((5, 4)), dtype=dtype)
+    Y = array_constr(5 * rng.random_sample((3, 4)), dtype=dtype)
+
+    try:
+        S = func(X, metric=metric, n_jobs=1, **kwds)
+    except (TypeError, ValueError) as exc:
+        # Not all metrics support sparse input
+        # ValueError may be triggered by bad callable
+        if array_constr is csr_matrix:
+            assert_raises(type(exc), func, X, metric=metric,
+                          n_jobs=2, **kwds)
+            return
+        else:
+            raise
+    S2 = func(X, metric=metric, n_jobs=2, **kwds)
+    assert_allclose(S, S2)
```

Comment:
`assert np.allclose(...)`, same below

Reply:
why not `assert_allclose` ? We tend to use `assert_allclose` more than `assert np.allclose`, and it comes from numpy.testing so it seems appropriate

Reply:
I opened https://github.com/scikit-learn/scikit-learn/issues/13180 but didn't get much traction.

I'm fine with either

Reply:
In that case I tend to prefer assert_allclose :)